### DISCUSSION TITLE
Possible to switch setting (e.g. bg) for all further cells/rows?

### DISCUSSION THREAD

**Discussion by FloMiLe** [lvjr/tabularray#412] [Q&A]
The commands ``\SetCell`` and ``\SetRow`` can be used to affect the settings of the cell/row they are placed in.

I am wondering whether there also is a command that changes a setting for all cells/rows after it. (If there is not, that's also fine, I would just like to know...)

**Comment by muzimuzhi** [comment]
Seems not. The old interface commands `\SetCells` and `\SetRows` influences the whole table.

But it's easy to define new table commands for these tasks:

```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}

\ExplSyntaxOn
\makeatletter
\NewTableCommand \SetFollowingRows [2] []
  {
    \tblr_set_following_rows:nn {#1} {#2}
  }

\cs_new_protected:Npn \tblr_set_following_rows:nn #1 #2
  {
    \group_begin:
    \int_step_inline:nnn { \c@rownum } { \c@rowcount }
      {
        \int_set:Nn \c@rownum {##1}
        \tblr_set_row:nn {#1} {#2}
      }
    \group_end:
  }

\NewTableCommand \SetFollowingCells [2] []
  {
    \tblr_set_following_cells:nn {#1} {#2}
  }

\cs_new_protected:Npn \tblr_set_following_cells:nn #1 #2
  {
    \group_begin:
    % cells on current row
    \int_step_inline:nnn { \c@colnum } { \c@colcount }
      {
        \int_set:Nn \c@colnum {##1}
        \tblr_set_cell:nn {#1} {#2}
      }
    % cells on next rows
    \int_step_inline:nnn { \c@rownum+1 } { \c@rowcount }
      {
        \int_set:Nn \c@rownum {##1}
        \int_step_inline:nn { \c@colcount }
          {
            \int_set:Nn \c@colnum {####1}
            \tblr_set_cell:nn {#1} {#2}
          }
      }
    \group_end:
  }
\makeatother
\ExplSyntaxOff

\begin{document}
\begin{tblr}{}
  a & b \\
  \SetRows{bg=red!20}
  c & d \\
  e & f
\end{tblr}
\quad
\begin{tblr}{}
  a & b \\
  \SetFollowingRows{bg=red!20}
  c & d \\
  e & f
\end{tblr}
\quad
\begin{tblr}{}
  a & b \\
  c & \SetFollowingCells{font=\bfseries, bg=gray, fg=white} d \\
  e & f
\end{tblr}
\end{document}
```
![image](https://github.com/lvjr/tabularray/assets/6376638/5b3fa999-c99b-4760-8f92-44f738a7e684)

**Reply by FloMiLe** [reply]
Thanks you so much! This works perfectly, and is much appreciated. 
(Programming this is rather impossible for me than easy. ;-) )

**Comment by muzimuzhi** [comment]
I realized that both `\SetRows` and `\SetCells` are part of the old interfaces (commands). If the new interfaces (inner style options) are preferable, then this means supporting single-side bounded ranges like `row{2-}` and `cells{-2}{3-5}` (but `cells{1-}{1-}` is ambiguous).

**Reply by lvjr** [reply]
No need to add new interfaces.
